### PR TITLE
fix(scylla_create_devices): improve logic for found devices

### PR DESCRIPTION
**Pull request by aleksbykov**
scylla_create_devices support 3 variants for data_device:
- auto: for aws instance will search nvme(ephemeral) device,
          if not found will search ebs volumes(ebs).
        for gce instances will search nvme(attached) devices
for aws instances:
- attached: will search ebs volumes(attached) and if such volumes
            added to os as nvme devices, will use them
- instance_store: will search nvme(ephemeral) devices and use

### CODE REVIEW

File: common/scylla_configure.py [hunk 1]
```diff
@@ -146,11 +155,10 @@ def start_scylla_on_first_boot(self):
...
 
     def create_devices(self):
-        device_type = self.instance_user_data.get("data_device")
-        use_device_type = "--data-device {}".format(device_type) if device_type else ""
-
+        device_type = self.instance_user_data.get("data_device", self.CONF_DEFAULTS['data_device'])
         try:
-            subprocess.run("/opt/scylladb/scylla-machine-image/scylla_create_devices {}".format(use_device_type), shell=True, check=True)
+            LOGGER.info(f"Create scylla data devices as {device_type}")
+            subprocess.run("/opt/scylladb/scylla-machine-image/scylla_create_devices --data-device {}".format(device_type), shell=True, check=True)
```
Comment (bentsi) [inline common/scylla_configure.py:161]: Use f-string:
```suggestion
            subprocess.run(f"/opt/scylladb/scylla-machine-image/scylla_create_devices --data-device {device_type}", shell=True, check=True)
```

File: common/scylla_create_devices [hunk 2]
```diff
@@ -36,12 +36,14 @@ def create_raid(devices):
...
-            return [os.path.join('/dev/', name) for name in instance.ebs_disks()]
-        elif device_type == "instance_store":
-            return [os.path.join('/dev/', name) for name in instance.ephemeral_disks()]
-        else:
-            raise Exception(f"Unknown devicetype {device_type}")
+            devices = [str(Path('/dev', name)) for name in instance.ebs_disks() if Path('/dev', name).exists()]
+        if not devices or device_type == "instance_store":
+            devices = [str(Path('/dev', name)) for name in instance.ephemeral_disks()]
+        if not devices:
+            raise Exception(f"No devices attached or instance_store were found devicetype")
```
Comment (bentsi) [inline common/scylla_create_devices:45]:
```suggestion
            raise Exception(f"No block devices were found for '{device_type}' device type")
```

**Review by bentsi — CHANGES_REQUESTED** [review]
1) please address comments
2) did you check that `instance.ebs_disks()` and `instance.ephemeral_disks()` returns correct values?

File: common/scylla_create_devices (hunk 2 above)
Comment (aleksbykov) [reply common/scylla_create_devices]: Done

File: common/scylla_configure.py (hunk 1 above)
Comment (aleksbykov) [reply common/scylla_configure.py]: Done

**Comment by aleksbykov** [comment]
> 1. please address comments

Done.
> 2. did you check that `instance.ebs_disks()` and `instance.ephemeral_disks()` returns correct values?

Yes, i had several instances with different configurations (ebs only as nvme, ebs only as attached, nvme+ebs) and i uploaded these files and check that disks correctly found and used. 
Also, i want to create new custom ami and check that scylla installed correctly

**Review by bentsi — APPROVED** [review]

**Comment by bentsi** [comment]
can you create AMI with this code and verify that this works?

**Comment by bentsi** [comment]
backported to 4.4

**Pull request merged**